### PR TITLE
Remove context from top level apply

### PR DESCRIPTION
**Pull request by serathius**
cc @ahrtr 

https://github.com/etcd-io/etcd/issues/18667#issuecomment-2393130782

### CODE REVIEW

File: server/etcdserver/apply/apply.go [hunk 1]
```diff
@@ -146,8 +146,8 @@ func newApplierV3Backend(
 		txnModeWriteWithSharedBuffer: txnModeWriteWithSharedBuffer}
 }
 
-func (a *applierV3backend) Apply(ctx context.Context, r *pb.InternalRaftRequest, applyFunc applyFunc) *Result {
-	return applyFunc(ctx, r)
+func (a *applierV3backend) Apply(r *pb.InternalRaftRequest, applyFunc applyFunc) *Result {
+	return applyFunc(context.TODO(), r)
```
Comment (ahrtr) [inline server/etcdserver/apply/apply.go:150]: Suggest to add similar comment as in https://github.com/etcd-io/etcd/pull/18674 right above this line to make this PR completely independent. In the future, when anyone continue to completely get rid of the context, then we can remove the comment by then.

**Review by ahrtr — APPROVED** [review]
LGTM with a minor comment.

Thanks for the change which looks clean.

File: server/etcdserver/apply/apply.go (hunk 1 above)
Comment (serathius) [reply server/etcdserver/apply/apply.go:150]: I have all commit ready. Just going one by one.

Comment (serathius) [reply server/etcdserver/apply/apply.go:150]: https://github.com/etcd-io/etcd/pull/18676 is next

**Pull request merged**